api/helpers/gameStates/deal-cards.js

```diff
@@ -0,0 +1,73 @@
+const GameStatus = require('../../../utils/GameStatus.json');
+const DeckIds = require('../../../utils/DeckIds.json');
+const MoveType = require('../../../utils/MoveType.json');
+const GamePhase = require('../../../utils/GamePhase.json');
+
+module.exports = {
+  friendlyName: 'Deal Cards',
+
+  description: 'Creates initial GameStateRow for specified game, with cards dealt ot both players (5 for p0 and 6 for p1). Errors if Game has previous game or incorrec status.',
+
+  inputs: {
+    game: {
+      type: 'ref',
+      description: 'Game record to be initialized with first GameState',
+      required: true,
+    },
+  },
+
+  fn: async ({ game }, exits) => {
+    
+    try {
+      if (game.status !== GameStatus.CREATED) {
+        return exits.error({ message: 'Game has already started' });
+      }
+
+      const previousGameStates = await GameStateRow.find({ gameId: game.id});
+      if (previousGameStates?.length) {
+        return exits.error({ message: 'Cards are already dealt' });
+      }
+
+      const deck = _.shuffle(
+        DeckIds.map(sails.helpers.gameStates.convertCardIdToCard)
+      );
+
+      const p0 = {
+        hand: deck.splice(0, 5),
```

Comment:
Pretty nifty to be able to call .splice() synchronously here to remove 5 cards from the deck and put them into the player hand in one line. This kind of move (and the ability to write it all to the db with a single insert and  function call) was a core motivator for the arch of the GameState API